### ISSUE TITLE
Release-0.33.x - Filter not inferring correct datatype for "Count of Rows" metric

### ISSUE THREAD

**Issue by UsmanYasin** [metabase/metabase#10552]
**Describe the bug**
When using BigQuery data source, if I apply filter on a pre-aggregated "Count of Rows" metric, the filter being applied is a text filter instead of a numeric filter resulting in "No matching signature for operator > for argument types: INT64, STRING. Supported signatures: ANY > ANY at [2:297]" error.

**Logs**
Nothing unusual in Frontend logs.
Backend Logs below for reference:
```
Aug 11 23:22:09 WARN metabase.util :: auto-retry metabase.driver.google$execute$fn__367@4a8d090b: No matching signature for operator > for argument types: INT64, STRING. Supported signatures: ANY > ANY at [2:297]
Aug 11 23:22:10 WARN metabase.util :: auto-retry metabase.driver.google$execute$fn__367@4a8d090b: No matching signature for operator > for argument types: INT64, STRING. Supported signatures: ANY > ANY at [2:297]
Aug 11 23:22:10 WARN metabase.util :: auto-retry metabase.driver.bigquery$process_native_STAR_$fn__671@47377aa6: No matching signature for operator > for argument types: INT64, STRING. Supported signatures: ANY > ANY at [2:297]
Aug 11 23:22:10 WARN metabase.util :: auto-retry metabase.driver.google$execute$fn__367@7f330bcb: No matching signature for operator > for argument types: INT64, STRING. Supported signatures: ANY > ANY at [2:297]
Aug 11 23:22:10 WARN metabase.util :: auto-retry metabase.driver.google$execute$fn__367@7f330bcb: No matching signature for operator > for argument types: INT64, STRING. Supported signatures: ANY > ANY at [2:297]
Aug 11 23:22:11 WARN metabase.query-processor.middleware.process-userland-query :: Query failure {:status :failed,
:class clojure.lang.ExceptionInfo,
:error "No matching signature for operator > for argument types: INT64, STRING. Supported signatures: ANY > ANY at [2:297]",
:stacktrace
("--> driver.google$execute_no_auto_retry.invokeStatic(google.clj:39)"
"driver.google$execute_no_auto_retry.invoke(google.clj:32)"
"driver.google$execute$fn__367.invoke(google.clj:50)"
"util$do_with_auto_retries.invokeStatic(util.clj:426)"
"util$do_with_auto_retries.invoke(util.clj:418)"
"util$do_with_auto_retries.invokeStatic(util.clj:430)"
"util$do_with_auto_retries.invoke(util.clj:418)"
"util$do_with_auto_retries.invokeStatic(util.clj:430)"
"util$do_with_auto_retries.invoke(util.clj:418)"
"driver.google$execute.invokeStatic(google.clj:49)"
"driver.google$execute.invoke(google.clj:43)"
"driver.bigquery$execute_bigquery.invokeStatic(bigquery.clj:174)"
"driver.bigquery$execute_bigquery.invoke(bigquery.clj:163)"
"driver.bigquery$execute_bigquery.invokeStatic(bigquery.clj:165)"
"driver.bigquery$execute_bigquery.invoke(bigquery.clj:163)"
"driver.bigquery$process_native_STAR_$fn__671.invoke(bigquery.clj:259)"
"util$do_with_auto_retries.invokeStatic(util.clj:426)"
"util$do_with_auto_retries.invoke(util.clj:418)"
"util$do_with_auto_retries.invokeStatic(util.clj:430)"
"util$do_with_auto_retries.invoke(util.clj:418)"
"driver.bigquery$process_native_STAR_.invokeStatic(bigquery.clj:258)"
"driver.bigquery$process_native_STAR_.invoke(bigquery.clj:254)"
"driver.bigquery$eval865$fn__867.invoke(bigquery.clj:443)"
"query_processor$fn__44642$execute_query__44647$fn__44648.invoke(query_processor.clj:70)"
"query_processor$fn__44642$execute_query__44647.invoke(query_processor.clj:64)"
"query_processor.middleware.mbql_to_native$mbql__GT_native$fn__35003.invoke(mbql_to_native.clj:38)"
"query_processor.middleware.annotate$result_rows_maps__GT_vectors$fn__36837.invoke(annotate.clj:537)"
"query_processor.middleware.annotate$add_column_info$fn__36743.invoke(annotate.clj:481)"
"query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__37764.invoke(cumulative_aggregations.clj:57)"
"query_processor.middleware.resolve_joins$resolve_joins$fn__41748.invoke(resolve_joins.clj:184)"
"query_processor.middleware.limit$limit$fn__38739.invoke(limit.clj:19)"
"query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__44511.invoke(results_metadata.clj:86)"
"query_processor.middleware.format_rows$format_rows$fn__38727.invoke(format_rows.clj:26)"
"query_processor.middleware.add_dimension_projections$add_remapping$fn__35412.invoke(add_dimension_projections.clj:234)"
"query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__36028.invoke(add_source_metadata.clj:101)"
"query_processor.middleware.resolve_source_table$resolve_source_tables$fn__41792.invoke(resolve_source_table.clj:48)"
"query_processor.middleware.add_row_count_and_status$add_row_count_and_status$fn__35891.invoke(add_row_count_and_status.clj:16)"
"query_processor.middleware.driver_specific$process_query_in_context$fn__37975.invoke(driver_specific.clj:12)"
"query_processor.middleware.resolve_driver$resolve_driver$fn__41412.invoke(resolve_driver.clj:23)"
"query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__37166$fn__37167.invoke(bind_effective_timezone.clj:9)"
"util.date$call_with_effective_timezone.invokeStatic(date.clj:88)"
"util.date$call_with_effective_timezone.invoke(date.clj:77)"
"query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__37166.invoke(bind_effective_timezone.clj:8)"
"query_processor.middleware.store$initialize_store$fn__44536$fn__44537.invoke(store.clj:11)"
"query_processor.store$do_with_store.invokeStatic(store.clj:46)"
"query_processor.store$do_with_store.invoke(store.clj:40)"
"query_processor.middleware.store$initialize_store$fn__44536.invoke(store.clj:10)"
"query_processor.middleware.async$async__GT_sync$fn__34914.invoke(async.clj:23)"
"query_processor.middleware.async_wait$runnable$fn__36894.invoke(async_wait.clj:89)"),
:query
{:query {:source-query {:source-table 11, :breakout [["field-id" 219]], :aggregation [["count"]]}, :filter [">" ["field-literal" "count" "type/Integer"] "1"], :limit 10},
:type "query",
:parameters [],
:async? true,
:middleware {:add-default-userland-constraints? true, :userland-query? true},
:info
{:executed-by 1,
:context :ad-hoc,
:card-id nil,
:nested? false,
:query-hash [119, -71, 91, 122, -69, -89, 111, 123, 77, 43, -61, 120, -12, 43, 31, -94, -65, -33, 87, -43, 29, -61, -33, -43, 102, 117, 87, 36, -20, -118, 26, 41]},
:constraints {:max-results 10000, :max-results-bare-rows 2000}},
:preprocessed
{:query
{:source-query {:source-table 11, :breakout [[:field-id 219]], :aggregation [[:aggregation-options [:count] {:name "count"}]], :order-by [[:asc [:field-id 219]]]},
:filter [:> [:field-literal "count" :type/Integer] [:value "1" nil]],
:limit 10,
:source-metadata
[{:name "customer_id",
:id 219,
:table_id 11,
:display_name "Customer ID",
:base_type :type/Integer,
:special_type nil,
:fingerprint
{:global {:distinct-count 184, :nil% 0.0},
:type {:type/Number {:min 9.30773598271E11, :max 1.234778587199E12, :avg 1.2326774316174854E12, :sd 2.1138953981871635E10, :q1 1.234181169215E12, :q3 1.2342048083637466E12}}},
:settings nil}
{:name "count", :display_name "Count", :base_type :type/Integer, :special_type :type/Number}],
:fields [[:field-literal "customer_id" :type/Integer] [:field-literal "count" :type/Integer]]},
:type :query,
:database 3,
:middleware {:add-default-userland-constraints? true, :userland-query? true},
:info
{:executed-by 1,
:context :ad-hoc,
:nested? false,
:query-hash [119, -71, 91, 122, -69, -89, 111, 123, 77, 43, -61, 120, -12, 43, 31, -94, -65, -33, 87, -43, 29, -61, -33, -43, 102, 117, 87, 36, -20, -118, 26, 41]},
:constraints {:max-results 10000, :max-results-bare-rows 2000},
:driver :bigquery},
:native
{:query
"SELECT `source`.`customer_id` AS `customer_id`, `source`.`count` AS `count` FROM (SELECT `shopel1f9yrou1n1qrt9hfmvrt8aq.orders_vw`.`customer_id` AS `customer_id`, count(*) AS `count` FROM `shopel1f9yrou1n1qrt9hfmvrt8aq.orders_vw` GROUP BY `customer_id` ORDER BY `customer_id` ASC) `source` WHERE `source`.`count` > '1' LIMIT 10",
:table-name :source,
:mbql? true},
:ex-data
{"code" 400,
"errors"
[{"domain" "global", "location" "q", "locationType" "parameter", "message" "No matching signature for operator > for argument types: INT64, STRING. Supported signatures: ANY > ANY at [2:297]", "reason" "invalidQuery"}],
"message" "No matching signature for operator > for argument types: INT64, STRING. Supported signatures: ANY > ANY at [2:297]",
"status" "INVALID_ARGUMENT"}}
```

**To Reproduce**
1. Add a BigQuery data source to your Metabase Instance
2. Create a custom question on the BigQuery data source
3. Group by on any column and set the metric to be "Count of Rows"
4. Apply Filter on the count column to be greater than 1
5. Visualize the result set

**Expected behavior**
The filter should be properly applied returning only groups with count more than 1

**Screenshots**
![image](https://user-images.githubusercontent.com/15028091/62840877-24de5400-bcba-11e9-93f1-c70b2442409f.png)
![image](https://user-images.githubusercontent.com/15028091/62840882-463f4000-bcba-11e9-9638-2e03b9c2d4ae.png)
![image](https://user-images.githubusercontent.com/15028091/62840888-5c4d0080-bcba-11e9-88db-4f6be14439e1.png)

**Information about your Metabase Installation:**
- Tried on multiple browser/OS environments
- Your databases: BigQuery
- Metabase version: The feature appears to be working fine on 0.32.x and 0.33 release preview  versions but appears to be broken in 0.33 branches(release-0.33.x and Release candidate branch)

- Metabase hosting environment: Docker

- Metabase internal database: MySQL

**Severity**
Frankly speaking, it is difficult for me to assign a severity level to this issue as it is annoying but at the same time can be bypassed by either using sql question or "count of distinct values"(partial workaround) metric.

**Comment by camsaul** [comment]
Looks like the issue with the MBQL query that gets generated

```clj
{:filter [">" ["field-literal" "count" "type/Integer"] "1"]}
```

It seems like in this case it would be possible to coerce the String filter value `"1"` to an integer automatically in the QP but we don't yet have that sort of logic. But either way I think we should disallow type-mismatched filter clauses on the FE too so you can't accidentally do something like

```clj
{:filter [">" ["field-literal" "count" "type/Integer"] "Not parseable as an integer"]}
```

**Comment by paulrosenzweig** [comment]
Fixed in #11800
